src/lib/utils/os_utils.cpp

```diff
@@ -614,6 +620,35 @@ void OS::page_named(void* page, size_t size) {
 #endif
 }
 
+#if defined(BOTAN_TARGET_OS_HAS_THREADS)
+void OS::set_thread_name(std::thread& thread, const char* const name) {
+   int r;
+   auto hdl = thread.native_handle();
+   #if defined(BOTAN_TARGET_OS_IS_LINUX)
+   r = pthread_setname_np(hdl, name);
+   #elif defined(BOTAN_TARGET_OS_IS_FREEBSD)
+   r = pthread_set_name_np(hdl, name);
+   #elif defined(BOTAN_TARGET_OS_IS_OPENBSD)
+   r = pthread_set_name_np(hdl, const_cast<char*>(name));
+   #elif defined(BOTAN_TARGET_OS_IS_NETBSD)
+   r = pthread_set_name_np(hdl, "%s", const_cast<char*>(name));
+   #elif defined(BOTAN_TARGET_OS_HAS_WIN32) && defined(BOTAN_BUILD_COMPILER_IS_MSVC)
+   // This approach (as opposed to trigger an exception) is simpler
+   // also gives a chance to be in coredumps.
+   wchar_t val[16];
+   if(MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS, name, -1, val, 16) > 0) {
+      r = SetThreadDescription(hdl, val);
```

Comment:
IIUC https://learn.microsoft.com/en-us/windows/win32/api/processthreadsapi/nf-processthreadsapi-setthreaddescription this function is only supported on Windows 10 or higher, which I think would be problematic

Reply:
In general the policy is that we don't support operating systems that are no longer supported by the OS vendor, but for Windows we are basically forced to be more lax than that simply due to the huge installed base of out of date systems.

I think practically speaking we are limited to using features that are available in the oldest version of Windows that is supported by the oldest version of VC++ we support. For VC that's VC2022 now. But I can't find anything definite about the oldest OS that VC2022 will allow you to target. I know we reverted #365 because people were still building XP-compatible binaries in 2016.